lms/djangoapps/program_enrollments/api/reading.py

```diff
@@ -377,6 +383,46 @@ def get_users_by_external_keys(program_uuid, external_user_keys):
     return users_by_external_keys
 
 
+def get_external_key_by_user_and_course(user, course_key):
+    """
+    Returns the external_user_key of the edX account/user
+    enrolled into the course
+
+    Arguments:
+        user (User):
+            The edX account representing the user in auth_user table
+        course_key (CourseKey|str):
+            The course key of the course user is enrolled in
+
+    Returns: external_user_key (str|None)
+        The external user key provided by Masters degree provider
+        Or None if cannot find edX user to Masters learner mapping
+    """
+    program_course_enrollments = ProgramCourseEnrollment.objects.filter(
+        course_enrollment__user=user,
+        course_key=course_key
+    ).order_by('-modified')
+
+    if not program_course_enrollments:
+        return None
+
+    program_course_enrollments_list = list(program_course_enrollments)
+    active_enrollment = (
+        pce for pce in program_course_enrollments_list
+        if pce.is_active
+    )
+
+    try:
+        # If we found there is an active program course enrollment, then according
+        # to the query order, this would be the most recent active record. we want
+        # to based on the external_user_key on this most recent active record.
+        # Otherwise, get the most recent course_enrollment
+        relevant_pce = next(active_enrollment or program_course_enrollments_list)
+        return getattr(relevant_pce.program_enrollment, 'external_user_key', None)
```

Comment:
`relevant_pce.program_enrollment` should never not have the `external_user_key` attribute defined. If it does, then we have a bigger problem on our hands, and we should probably just let it `AttributeError` out :)